libraries/ms-common/inc/extra_unity.h

```diff
@@ -1,5 +1,11 @@
 #pragma once
+
+#include <stdint.h>
+
 #include "status.h"
 #include "unity.h"
 
 #define TEST_ASSERT_OK(code) TEST_ASSERT_EQUAL(STATUS_CODE_OK, (code))
+
+// TEST ONLY FUNCTION TO SET TIMER COUNTER FOR SOFT TIMERS UNSAFE TO CALL OUTSIDE A TEST.
+void TEST_SOFT_TIMER_SET_COUNTER(uint32_t counter_value);
```

Comment:
This should probably not be all caps. I think this file should be renamed to test_helpers or something more generic.

I'm also not sure if these test functions should be separated from their modules. We can always have a [module]_(test|priv|impl).h header that includes test-only/"private" functions.